### PR TITLE
feat/add question counter

### LINKED ISSUE #7: [Feature] Add questions counter
**Is your feature request related to a problem? Please describe.**
Currently the application has nothing to show how many questions the user has answered.
A counter could be useful.

**Describe the solution you'd like**
Add a counter inside the form, above the question. (eg. 1/480)

image

### PR DESCRIPTION
**Pull request by Lucif3r-in**
Fixes #7 

- [x] I have read and agree to the [Code of Conduct](https://github.com/eduardconstantin/azure-fundamentals/blob/main/CODE_OF_CONDUCT.md), and have followed the [Contributing guidelines](https://github.com/eduardconstantin/azure-fundamentals/blob/main/CONTRIBUTING.md).

## Pull Request Type
- [x] New feature

### CODE REVIEW

**Comment by Lucif3r-in** [comment]
@eduardconstantin Please review it!

File: components/QuizForm.tsx [hunk 1]
```diff
@@ -30,10 +42,29 @@ const QuizForm: React.FC<Props> = ({ isLoading, questionSet, handleNextQuestion
...
+  const handleNext = () => {
+    reset();
+    setShowCorrectAnswer(false);
+    setQuestionsSolved(questionsSolved + 1);
+    handleNextQuestion();
+  };
+
   return (
-    <form onSubmit={handleSubmit(onSubmit)}>
+    <div className="relative">
```
Comment (eduardconstantin) [inline components/QuizForm.tsx:53]: Don't remove the form tag
```suggestion
    <form onSubmit={handleSubmit(onSubmit)}>
```

File: components/QuizForm.tsx [hunk 2]
```diff
@@ -51,23 +82,25 @@ const QuizForm: React.FC<Props> = ({ isLoading, questionSet, handleNextQuestion
...
-            setShowCorrectAnswer(false);
-            handleNextQuestion();
-          }}
+          onClick={handleNext}
         >
           Next Question
         </Button>
       </div>
-    </form>
+    </div>
```
Comment (eduardconstantin) [inline components/QuizForm.tsx:103]:
```suggestion
    </form>
```

File: components/QuizForm.tsx [hunk 3]
```diff
@@ -30,10 +42,29 @@ const QuizForm: React.FC<Props> = ({ isLoading, questionSet, handleNextQuestion
     setShowCorrectAnswer(true);
   };
 
+  const handleNext = () => {
+    reset();
+    setShowCorrectAnswer(false);
+    setQuestionsSolved(questionsSolved + 1);
```
Comment (eduardconstantin) [inline components/QuizForm.tsx:48]: you can remove this too

File: components/QuizForm.tsx [hunk 4]
```diff
@@ -11,15 +11,27 @@ type Props = {
...
-  const { register, handleSubmit, reset } = useForm();
+const QuizForm: React.FC<Props> = ({
+  isLoading,
+  questionSet,
+  handleNextQuestion,
+}) => {
+  const { register, reset } = useForm();
   const [showCorrectAnswer, setShowCorrectAnswer] = useState<boolean>(false);
   const [windowWidth, setWindowWidth] = useState<number>(0);
+  const [questionsSolved, setQuestionsSolved] = useState<number>(0);
```
Comment (eduardconstantin) [inline components/QuizForm.tsx:22]: you can remove this state, you have one in Quiz component, use that counter

File: components/QuizForm.tsx [hunk 5]
```diff
@@ -11,15 +11,27 @@ type Props = {
   handleNextQuestion: () => void;
 };
 
-const QuizForm: React.FC<Props> = ({ isLoading, questionSet, handleNextQuestion }) => {
-  const { register, handleSubmit, reset } = useForm();
+const QuizForm: React.FC<Props> = ({
+  isLoading,
+  questionSet,
+  handleNextQuestion,
```
Comment (eduardconstantin) [inline components/QuizForm.tsx:17]: Add a prop here for the question counter

File: components/QuizForm.tsx [hunk 6]
```diff
@@ -11,15 +11,27 @@ type Props = {
...
+}) => {
+  const { register, reset } = useForm();
   const [showCorrectAnswer, setShowCorrectAnswer] = useState<boolean>(false);
   const [windowWidth, setWindowWidth] = useState<number>(0);
+  const [questionsSolved, setQuestionsSolved] = useState<number>(0);
+  // const [totalQuestions, setTotalQuestions] = useState<number>(0);
```
Comment (eduardconstantin) [inline components/QuizForm.tsx:23]: please remove comments

File: components/QuizForm.tsx [hunk 7]
```diff
@@ -51,23 +82,25 @@ const QuizForm: React.FC<Props> = ({ isLoading, questionSet, handleNextQuestion
         ))}
       </ul>
       <div className="flex justify-center flex-col sm:flex-row">
-        <Button type="submit" intent="secondary" size="medium">
+        <Button
+          type="submit"
+          intent="secondary"
+          size="medium"
+          onClick={onSubmit}
```
Comment (eduardconstantin) [inline components/QuizForm.tsx:89]: remove this onClick

File: components/types.ts [hunk 8]
```diff
@@ -1,4 +1,5 @@
 export type Question = {
+  // length(length: any): unknown;
```
Comment (eduardconstantin) [inline components/types.ts:2]:
```suggestion

```

File: components/QuizForm.tsx [hunk 9]
```diff
@@ -30,10 +42,29 @@ const QuizForm: React.FC<Props> = ({ isLoading, questionSet, handleNextQuestion
...
+    setQuestionsSolved(questionsSolved + 1);
+    handleNextQuestion();
+  };
+
   return (
-    <form onSubmit={handleSubmit(onSubmit)}>
+    <div className="relative">
+      <div className="flex justify-center items-center mb-2">
+        <p className="text-white font-medium text-sm">
+          {questionsSolved + 1} / 480
```
Comment (eduardconstantin) [inline components/QuizForm.tsx:56]: add here the prop
```suggestion
          {questionNo + 1} / 480
```

**Review by eduardconstantin — CHANGES_REQUESTED** [review]

**Comment by eduardconstantin** [comment]
you can use this state from Quiz component as a counter, no need to make a new state inside the QuizForm component
image

**Comment by Lucif3r-in** [comment]
@eduardconstantin Done!!

**Review by eduardconstantin — APPROVED** [review]
Thank you for helping me with this issue!

**Comment by Lucif3r-in** [comment]
> Thank you for helping me with this issue!

Thank you for guiding me!! Have a good day!!

**Pull request merged**